cdc/listener.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdc
+
+import (
+	"errors"
+
+	"github.com/apple/foundationdb/bindings/go/src/fdb"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/tigrisdata/tigrisdb/internal"
+)
+
+type Tx struct {
+	Ops []Op
+}
+
+type Op struct {
+	Type  string
+	Table []byte
+	Key   []byte `json:",omitempty"`
+	LKey  []byte `json:",omitempty"`
+	RKey  []byte `json:",omitempty"`
+	Data  []byte `json:",omitempty"`
+}
+
+func (tx *Tx) addOp(entry Op) {
+	tx.Ops = append(tx.Ops, entry)
+}
+
+type TxListener struct {
```

Comment:
if you do this way then we still have only one listener? Don't we need to have TxLister inside the Publisher?

Reply:
a `TxListener` is created per context, when it's wrapped